core/node/external_proof_integration_api/src/processor.rs

```diff
@@ -45,17 +69,32 @@ impl Processor {
     pub(crate) async fn verify_proof(
         &self,
         Path(l1_batch_number): Path<u32>,
-        Json(payload): Json<VerifyProofRequest>,
+        mut multipart: Multipart,
     ) -> Result<(), ProcessorError> {
-        let mut guard = MethodCallGuard::new(Method::VerifyProof);
-
         let l1_batch_number = L1BatchNumber(l1_batch_number);
         tracing::info!(
             "Received request to verify proof for batch: {:?}",
             l1_batch_number
         );
 
-        let serialized_proof = bincode::serialize(&payload.0)?;
+        let mut serialized_proof = vec![];
+
+        while let Some(field) = multipart
+            .next_field()
+            .await
+            .map_err(|_| ProcessorError::InvalidProof)?
+        {
+            if field.name() == Some("proof")
+                && field.content_type() == Some("application/octet-stream")
+            {
+                serialized_proof.extend_from_slice(&field.bytes().await.unwrap());
+            }
+        }
+
+        tracing::info!("Received proof is size: {}", serialized_proof.len());
+
+        let payload: VerifyProofRequest = bincode::deserialize(&serialized_proof)?;
+
         let expected_proof = bincode::serialize(
```

Comment:
Pass-by thought: proofs can be wiped from blob store, right? We want to report such cases gracefully, so if the error is "not found" we probably want to check if such a batch exists. If it does -- return `ProofWasPurged` error, otherwise, `ProofNotFound`. Can be tackled separately.

Reply:
It is done in a bit different way, I have added one more tackling in verify_proof, but generally - it will return object store error with message that batch is too old. Otherwise, if the batch number is bigger than latest proven - it will return BatchNotReady error.